conformance/utils/suite/features.go

```diff
@@ -169,6 +173,7 @@ var MeshCoreFeatures = sets.New(
 // Features - Experimental
 // -----------------------------------------------------------------------------
 
+// TODO(liorlieberman) get consensus about making experimental features per resource.
```

Comment:
I think I'd rather just tie this specific feature to Gateway since I think all test coverage we have for this also requires Gateway support.

Reply:
tying this to an HTTPRoute as per our offline discussion and as we only have conformance test that checks the HTTPRoute functionality with Destination Port Matching.